pkg/controllers/membercluster/v1beta1/membercluster_controller.go

```diff
@@ -452,6 +452,14 @@ func (r *Reconciler) syncInternalMemberClusterStatus(imc *clusterv1beta1.Interna
 	r.aggregateJoinedCondition(mc)
 	// Copy resource usages.
 	mc.Status.ResourceUsage = imc.Status.ResourceUsage
+	// Copy additional conditions.
+	for idx := range imc.Status.Conditions {
+		cond := imc.Status.Conditions[idx]
+		cond.ObservedGeneration = imc.GetGeneration()
```

Comment:
we always use the generation of mc itself when building the condition.

Not sure if it's valid for ClusterPropertyProviderStarted condition

Reply:
Fixed.